pytest_picked.py

```diff
@@ -64,15 +68,20 @@ def _affected_tests():
     """
     raw_output = _get_git_status()
 
+    re_list = [item.replace('.', '\.').replace('*', '.*')
+               for item in test_files]
+    re_string = "|".join(re_list)
+
     folders, files = [], []
     for candidate in raw_output.splitlines():
         file_or_folder = _extract_file_or_folder(candidate)
 
-        if "test" in file_or_folder:
-            if file_or_folder.endswith("/"):
+        if file_or_folder.endswith("/"):
+            path = Path(file_or_folder)
+            if any([part.startswith("test") for part in path.parts]):
```

Comment:
In this case, I don't think we need to check if the folder has `test` on it. For instance, if you add a new folder called `api` and inside of it you have other folders or `test_bla.py`, we are going to miss it. `git status` will show just `api/`. IMO, for this case, we must rely on `pytest` to check the files.

Reply:
Couldn't we check inside the new folder for test files? This is what I meant by getting rid of folder checking and just checking the flles.
Where `$ git status --short` will return:
```
 M project/accounts/tests/test_api.py
?? newapp/
```

$ git status --short --untracked-files' will return:
```
 M project/accounts/tests/test_api.py
?? newapp/__init__.py
?? newapp/admin.py
?? newapp/apps.py
?? newapp/migrations/__init__.py
?? newapp/models.py
?? newapp/tests/test_views.py
?? newapp/views.py
```
So that we can just check by the file names. See: https://git-scm.com/docs/git-status/1.8.1#git-status---untracked-filesltmodegt

Oddly enough, whenever an untracked folder is added, git status immediately shows the individual files.

Reply:
Couldn't we check inside the new folder for test files? This is what I meant by getting rid of folder checking and just checking the flles.
Where `$ git status --short` will return:
```
 M project/accounts/tests/test_api.py
?? newapp/
```

$ git status --short --untracked-files' will return:
```
 M project/accounts/tests/test_api.py
?? newapp/__init__.py
?? newapp/admin.py
?? newapp/apps.py
?? newapp/migrations/__init__.py
?? newapp/models.py
?? newapp/tests/test_views.py
?? newapp/views.py
```
So that we can just check by the file names. see: https://git-scm.com/docs/git-status/1.8.1#git-status---untracked-filesltmodegt

Oddly enough, whenever an untracked folder is added, git status immediately shows the individual files.

Perhaps we address this in a separate ticket? For the meantime, I've changed it as requested.